### PR TITLE
#168216140 Setup PostgreSQL with Docker

### PR DESCRIPTION
**Pull request by Oliver-ke**
#### Description of Task to be completed?
- add Postgres service to docker.compose.yml file
- link existing node app to use the new Postgres service
- test container using `docker-compose up` command

 #### How should this be manually tested?
- from develop run `git fetch --all` to get all recent changes
- run `git checkout ch-setup-docker-postgres-168216140`
- start-up the container by running `docker-compose up` or `sudo docker-compose up` at the project root, you should see the image building
- after a successful build, you should see the server running on `localhost:3000`
- also, you could be able to run commands such as `npm run test` or `npm run migrate` on the container by starting the container with `docker-compose run app bash` or `sudo docker-compose run app bash`

 #### Any background context you want to provide?
- ensure you have docker and docker-compose working
- ensure you have an internet connection, needed to pull the base images 

 #### What are the relevant pivotal tracker stories?
[#168216140](https://www.pivotaltracker.com/story/show/168216140)

### CODE REVIEW

File: src/test/controllers/users.test.js [hunk 1]
```diff
@@ -90,3 +91,21 @@ describe('User Registration test', () => {
...
+    const res = {
+      status: () => ({ json: jsonVal => JSON.stringify(jsonVal) }),
+    };
+    // call controller with mocked req,res objects
+    let returnValue = await userController(req, res);
+    returnValue = JSON.parse(returnValue);
+    chai.expect(returnValue.status).equal(500);
+    chai.expect(returnValue).to.have.property('message');
+  });
+});
```
Comment (oyewoas) [inline src/test/controllers/users.test.js:132]: well done @Oliver-ke , please what does this test do??

Comment (Oliver-ke) [reply src/test/controllers/users.test.js:132]: Thanks @oyewoas , While completing the task some section of the code in users controller which was the catch block was not tested so code climate was erroring, I had to write a test for that block. Here I'm mocking a request that should throw an error and then I'm using chai expect to see if an error was truly returned

Comment (oyewoas) [reply src/test/controllers/users.test.js:132]: ok @Oliver-ke

File: docker-compose.yaml [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+      PORT: 3000
     ports:
       - 3000:3000
     command: npm run start:dev
     volumes:
       - .:/app/
       - /app/node_modules
+
+  postgres:
+    image: postgres:10.4
```
Comment (sa-ma) [inline docker-compose.yaml]: Any specific reason for going with version 10.4 of postgres and not the latest

Comment (Oliver-ke) [reply docker-compose.yaml]: Ok @sa-ma , No much specific reason, but rather based on best practices, we need to be version-specific, doing this prevents the app from breaking whenever a new version different from the version used while building the app is released. At the long run, when new versions are released some functions on which our app depends on might become obsolete leading to the breakdown of the app

Comment (sa-ma) [reply docker-compose.yaml]: No problem, rebase so I can approve there are conflicts

Comment (Oliver-ke) [reply docker-compose.yaml]: Done

**Review by sa-ma — DISMISSED** [review]

**Pull request merged**